### ISSUE TITLE
Figure out which Sphere v2 APIs can be used in Cell

### ISSUE THREAD

**Issue by fatcerberus** [spheredev/neosphere#91]
In order to make Cell accessible to those already familiar with Sphere development, the Cellscript API should be as similar to Sphere v2 as possible.  To that end, it may be possible to copy over some calls verbatim.

Core API:
- [x] `system`, at least `.name` and `.version`
- [x] `FS`
- [x] `RNG`
- [x] `TextEncoder`/`TextDecoder` (provided by Duktape)

Standard Library:
- [x] `assert` module
- [x] `from` module
- [x] `random` module
- [x] `struct` module
- [x] `xml` module

**Comment by fatcerberus** [comment]
The utility of `RNG` is debatable in a build tool, which needs to be deterministic.  The only thing I can think of would be generating a unique identifier for each build, but that can be done internally, no need for an API.

Even so, `from()` relies on the "random" module, which needs `RNG`.  And I'd prefer not to maintain two versions of the same module.  Hm...

**Comment by fatcerberus** [comment]
I decided to bite the bullet and implement `RNG` in Cell.  `xoroshiro.c` is already in `src/shared`, so it's just a matter of copying the `RNG` implementation over from minisphere.

**Comment by fatcerberus** [comment]
I'll review the Sphere v2 API one last time to make sure I didn't miss anything useful, but this should be pretty much done now.

**Comment by Eggbertx** [comment]
Did you fix the issue with Color.of being undefined?

**Comment by fatcerberus** [comment]
Yes, that was fixed a while back and should work again once I release minisphere 4.4.

But this issue was actually about what part of Sphere v2 should be duplicated for use by Cell build scripts, not the engine itself.

**Comment by fatcerberus** [comment]
I added the `xml` module to `#/cell_modules`.  XML parsing may come in handy in Cellscripts.  Other than that, I looked through both the Sphere v2 core API and miniRT and the rest are pretty specific to the engine.

One thing that might be useful to add the `Socket` and `Server` classes to the Cellscript API, to allow the compiler network access for, e.g., downloading files at build time.  But that's a pretty low-level API for a build tool, so I'm not sure.  I'll keep it in mind for 5.0.0.
